src/bgw/job.c

```diff
@@ -1300,22 +1304,42 @@ ts_bgw_job_set_job_entrypoint_function_name(char *func_name)
 	job_entrypoint_function_name = func_name;
 }
 
+/*
+ * Run job and set next start.
+ *
+ * job: Job to run and update
+ * func: Function to execute for the job
+ * initial_runs: Limit on the number of runs to do
+ * next_interval: Interval to use when computing next start
+ * atomic: Should be executed as a single transaction.
+ * mark: Mark the start and end of the function execution in job_stats
+ */
 bool
 ts_bgw_job_run_and_set_next_start(BgwJob *job, job_main_func func, int64 initial_runs,
```

Comment:
note that this method is `only` used to execute the "telemetry" job...I'm not sure if we want to add/retain and have the atomic; mark / etc features for no good reason...as there doesn't seem to be a usecase for it...

only place its used outside is in a mock...

Reply:
> note that this method is `only` used to execute the "telemetry" job...I'm not sure if we want to add/retain and have the atomic; mark / etc features for no good reason...as there doesn't seem to be a usecase for it...
> 
> only place its used outside is in a mock...

Since it is used by the mock, it does not make sense to do a refactoring here and I prefer to handle this as a separate PR.

Reply:
I separated it into two flags `mark` and `atomic` mostly for stylish issues. The first implementation piggybacked this, but wanted to avoid confusing the situation.

One option is to rename this function to something more succinct with a focus on usage only for the telemetry job.